server/src/test/java/org/opensearch/extensions/action/ExtensionHandleTransportRequestTests.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.io.stream.BytesStreamInput;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.nio.charset.StandardCharsets;
+
+public class ExtensionHandleTransportRequestTests extends OpenSearchTestCase {
+    public void testExtensionHandleTransportRequest() throws Exception {
+        String action = "test-action";
+        byte[] requestBytes = "request-bytes".getBytes(StandardCharsets.UTF_8);
+        ExtensionHandleTransportRequest request = new ExtensionHandleTransportRequest(action, requestBytes);
+
+        assertEquals(action, request.getAction());
+        assertEquals(requestBytes, request.getRequestBytes());
+
+        BytesStreamOutput out = new BytesStreamOutput();
+        request.writeTo(out);
+        BytesStreamInput in = new BytesStreamInput(BytesReference.toBytes(out.bytes()));
+        request = new ExtensionHandleTransportRequest(in);
+
+        assertEquals(action, request.getAction());
+        assertEquals(new String(requestBytes, StandardCharsets.UTF_8), new String(request.getRequestBytes(), StandardCharsets.UTF_8));
```

Comment:
And a third comparison.  Looking at this again (3x) I'm thinking if you want to do a string comparison you should keep the original "request-bytes" string and compare that to a new String(getRequestBytes).   Or just compare arrays. But don't re-create the string you already had.